src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const [isLoading, setIsLoading] = useState(false);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(currentTodo?.userId || 0)
+      .then(setCurrentUser)
+      .catch(err => {
+        throw err;
+      })
```

Comment:
Throwing an error inside a promise rejection handler can lead to unhandled promise rejections. Consider logging the error or displaying a user-friendly message instead.